### PR TITLE
refactor(Filter): tailwindify

### PR DESCRIPTION
**Pull request by julio8a**
Supports #1658 #1500

### CODE REVIEW

File: src/components/calcite-filter/calcite-filter.scss [hunk 1]
```diff
@@ -1,66 +1,61 @@
...
-  display: flex;
-  margin: 0 var(--calcite-spacing-quarter);
-  overflow: hidden;
-  position: relative;
-  width: 100%;
+  @apply flex 
+    items-center 
+    overflow-hidden 
+    relative w-full;
+  margin: 0 theme("spacing.1");
```
Comment (paulcpederson) [inline src/components/calcite-filter/calcite-filter.scss:13]: `mx-1 my-0` maybe here

Comment (paulcpederson) [reply src/components/calcite-filter/calcite-filter.scss:13]: IDK @asangma can you explain the `--calcite-spacing-quarter` you probably had a reason for that? Or is it just old?

Comment (julio8a) [reply src/components/calcite-filter/calcite-filter.scss:13]: @asangma  ^

Comment (julio8a) [reply src/components/calcite-filter/calcite-filter.scss]: FYI, I moved the margin to `@apply` @paulcpederson

Comment (julio8a) [reply src/components/calcite-filter/calcite-filter.scss]: From a chat with Alan, we can get rid of all the %compoent-spacing vars.

**Review by paulcpederson — APPROVED** [review]

**Pull request merged**